synapse/storage/data_stores/main/registration.py

```diff
@@ -911,6 +911,7 @@ def __init__(self, database: Database, db_conn, hs):
         super(RegistrationStore, self).__init__(database, db_conn, hs)
 
         self._account_validity = hs.config.account_validity
+        self._ignore_unknown_session_error = hs.config.request_token_inhibit_3pid_errors
```

Comment:
It would be nice to add a unit test for this endpoint. There's an existing test for this config option at `test_password_reset_bad_email_inhibit_error`.

It'd be nice to check that session ID errors aren't returned when giving this endpoint an invalid session ID when this option is enabled. Unfortunately, it might be a little bit hard to check for that specific error are all errors here are `ThreepidValidationError 400`s...

Reply:
It'd probably be good enough with a unit test for `validate_threepid_session`, which is probably what I'm going to do.